### PR TITLE
MNT Use React Testing Library

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/silverstripe-admin/issues/1503

### CODE REVIEW

File: client/src/components/HistoryViewer/tests/HistoryViewerVersionState-test.js [hunk 1]
```diff
@@ -1,120 +1,109 @@
...
-      expect(component.getBadges()).toBe('');
-    });
-  });
+      },
+    })}
+    />
+  );
+  const badge = container.querySelector('.test-badge');
+  expect(badge.getAttribute('data-message')).toBe('Live');
+  expect(badge.getAttribute('data-status')).toBe('success');
```
Comment (sabina-talipova) [inline client/src/components/HistoryViewer/tests/HistoryViewerVersionState-test.js:108]: Do we need to test that "Returns an empty string when version is not live"?

File: package.json [hunk 2]
```diff
@@ -59,19 +54,16 @@
...
-    "react-16": "npm:react@^16.14.0",
-    "react-dom-16": "npm:react-dom@^16.14.0",
     "terser-webpack-plugin": "^5.3.6",
     "webpack": "^5.74.0",
     "webpack-cli": "^5.0.0"
   },
   "dependencies": {
     "@apollo/client": "^3.7.1",
     "@popperjs/core": "^2.11.6",
+    "@testing-library/react": "^14.0.0",
```
Comment (sabina-talipova) [inline package.json]: Should we move `testing-library/react` to the `devDependencies` section?

File: client/src/components/HistoryViewer/tests/HistoryViewerVersionState-test.js (hunk 1 above)
Comment (emteknetnz) [reply client/src/components/HistoryViewer/tests/HistoryViewerVersionState-test.js:108]: Added test

File: package.json (hunk 2 above)
Comment (emteknetnz) [reply package.json]: Updated

**Review by sabina-talipova — APPROVED** [review]
LGTM
